### DISCUSSION TITLE
Support for IDT with RGB input?

### DISCUSSION THREAD

**Discussion by priikone** [colour-science/colour#795] [Ideas]
Would it make sense to do a version of matrix_idt() that takes as input the linear camera RGB values (for a target) instead of the spectral sensitivities (since they're not always known)?  This would be the P-2013-001 Appendix B procedure.  It seems quite straightforward to do, unless I'm missing something in the current code.  I was wondering about the illuminant normalization which is done also for the training_data_sds_to_XYZ().  Is that actually necessary?  I tested without it and got identical results down to 7-8 digits.  The other thing to wonder is the white balance multipliers.  Maybe the input should always be white balanced?

Other thing I've been thinking is recovery of spectral sensitivities from a colour target image.  I've seen couple interesting papers around with nice results.  I'm still learning and trying to understand all these concepts so I probably don't know enough to implement these, but that would be a nice feature to have.  It would be interesting to compare the resulting 3x3 matrices optimized with measured sensitivities, recovered sensitivities and photographed colour target and see how much the three matrices actually differ.

"Recovery of Spectral Sensitivity Functions from a Colour Chart Image under Unknown Spectrally Smooth Illumination"
http://citeseerx.ist.psu.edu/viewdoc/download?doi=10.1.1.700.6064&rep=rep1&type=pdf

"What is the Space of Spectral Sensitivity Functions for Digital Color Cameras"
https://www.gujinwei.org/research/camspec/

**Comment by thomasmansencal** [comment]
Hello @priikone,

So as I'm participating in the authoring of the Academy IDT Calculator, we are actually implementing the IDT in various languages, e.g. CTL, DCTL and Nuke. As part of that effort, the `colour.characterisation.aces_it` module has received some updates and continue to do so if you look here: https://github.com/colour-science/colour/tree/feature/idt

> I was wondering about the illuminant normalization which is done also for the training_data_sds_to_XYZ(). Is that actually necessary?

Good question! I haven't looked at the code in a while!

> The other thing to wonder is the white balance multipliers. Maybe the input should always be white balanced?

Sorry, I'm not sure to understand, what input are you talking about here? The camera RGB values?

> Other thing I've been thinking is recovery of spectral sensitivities from a colour target image. I've seen couple interesting papers around with nice results.

I have on my todo list the last one to implement, it should not be too hard!

Adding @aforsythe and @nick-shaw for vis!

**Reply by nick-shaw** [reply]
Only had a quick scan read of the paper on spectral sensitivity estimation from a ColorChecker. Interesting concept. But I notice it mentions an assumption of Luther condition, which is not the case for any actual camera, is it?

**Reply by priikone** [reply]
> Sorry, I'm not sure to understand, what input are you talking about here? The camera RGB values?

Yes, I was wondering about computing the white balance multipliers in this case.

I'll put a commit together so you'll see what I'm talking about...

**Reply by priikone** [reply]
Here's a commit that gives an idea:  https://github.com/priikone/colour/commit/4ab53688a4ecf6ef244ab659e83e3cdea559ec49

Unit tests are missing but I tested with a target image and the resulting matrix was sane looking... But I didn't actually test with it yet so very much work-in-progress.

**Reply by priikone** [reply]
> 
> 
> Only had a quick scan read of the paper on spectral sensitivity estimation from a ColorChecker. Interesting concept. But I notice it mentions an assumption of Luther condition, which is not the case for any actual camera, is it?

Indeed.  The second paper plots the deviation from the Luther condition and as expected they all deviate.  So I assume they just ignore it.
